File: src/sorting/exchangeSort.c
#include <stdio.h>
#include <stdlib.h>
#include "../_common_/common.h"

void exchange_sort(int* arr, const int size)
{
    for (int i = 0; i < size - 1; i++)
        for (int j = i + 1; j < size; j++)
            if (arr[i] > arr[j]) swap(&arr[i], &arr[j]);
}

int main(void)
{
    int n;

    scanf("%d", &n);
    int* arr = (int*)malloc(sizeof(int) * n);
    for (int i = 0; i < n; i++)
        scanf("%d", &arr[i]);
    print_array(arr, n);
    exchange_sort(arr, n);
    print_array(arr, n);
    
    free(arr);
    return 0;
}

/*
Input:
5
5 4 3 2 1

Output:
5 4 3 2 1
1 2 3 4 5
*/

File: src/sorting/exchangeSort.py
def exchange_sort(arr: list, n: int) -> None:
    for i in range(0, n - 1):
        for j in range(i + 1, n):
            if (arr[i] > arr[j]):
                arr[i], arr[j] = arr[j], arr[i]


def print_array(arr: list, n: int) -> None:
    for i in range(0, n):
        print(arr[i], end=' ')
    print()


arr = list(map(int, input().split()))
size = len(arr)

print_array(arr, size)
exchange_sort(arr, size)
print_array(arr, size)

'''
Input:
5 4 3 2 1

Output:
5 4 3 2 1
1 2 3 4 5
'''